### ISSUE TITLE
Handling crashing drones in simulation controller

### ISSUE THREAD

**Issue by ZGako** [AP-2024-25-D-R-O-N-E/ap-project#16]
When a drone crashes it is removed from the network, but the way it's setup right now, when a drone receives a crash command, it may still have to handle various packets before it actually crashes.
Since this is still information worth showing, we need to be able to tell when a drone actually finishes executing to then remove it from the SC.

I think we can use the join_handle function is_finished to determine when this is the case.
Basically when a crash command is sent, the simulation controller will check for is_finished on that drone's thread every time it receives a PacketSent event from it
